pkg/controller/hosted/controller.go

```diff
@@ -238,16 +241,33 @@ func (r *ReconcileHosted) deleteManifestWorks(ctx context.Context, cluster *clus
 		return helpers.ForceDeleteAllManifestWorks(ctx, r.clientHolder.RuntimeClient, r.recorder, append(works, hostedWorks...))
 	}
 
-	// delete works that do not include klusterlet addon works, the addon works will be removed by
-	// klusterlet-addon-controller, we need to wait the klusterlet-addon-controller delete them
+	// delete works that do not include klusterlet works and klusterlet addon works, the addon works was removed
+	// above, we need to wait them to be deleted.
 	ignoreAddons := func(clusterName string, manifestWork workv1.ManifestWork) bool {
-		return strings.HasPrefix(manifestWork.GetName(), fmt.Sprintf("%s-klusterlet-addon", manifestWork.GetNamespace()))
+		manifestWorkName := manifestWork.GetName()
+		switch {
+		case strings.HasPrefix(manifestWorkName, fmt.Sprintf("%s-klusterlet-addon", manifestWork.GetNamespace())):
+		case strings.HasPrefix(manifestWorkName, "addon-") && strings.HasSuffix(manifestWork.GetName(), "-deploy"):
+		case strings.HasPrefix(manifestWorkName, "addon-") && strings.HasSuffix(manifestWork.GetName(), "-pre-delete"):
+		default:
+			return false
+		}
+		return true
 	}
 	err := helpers.DeleteManifestWorkWithSelector(ctx, r.clientHolder.RuntimeClient, r.recorder, cluster, works, ignoreAddons)
 	if err != nil {
 		return err
 	}
 
+	noAddons, err := helpers.NoManagedClusterAddons(ctx, r.clientHolder.RuntimeClient, cluster.GetName())
+	if err != nil {
+		return err
```

Comment:
I have not take a deep look at this code before, but we should consider requeue a certain time instead of error fallback if there are still returning addon or works. We might not need to do this in this PR, but we should consider it later 

cc @zhiweiyin318